docker_templates/templates/snippet/setup_ros_sync_date.Dockerfile.em

```diff
@@ -0,0 +1,17 @@
+@{
+import json
+import urllib.request
+ 
+# Define the URL
+url = f"http://api.github.com/repos/ros/rosdistro/git/matching-refs/tags/{ros_distro}"
+ 
+# Fetch the data
+with urllib.request.urlopen(url) as response:
+    data = json.loads(response.read().decode())
+ 
+# Sort the entries by the "ref" value
+sorted_data = sorted(data, key=lambda x: x['ref'])
+
+ros_sync_date = sorted_data[-1]['ref'].split('/')[-1]
+}@
+ENV ROS_SYNC_DATE=@(ros_sync_date)@
```

Comment:
This was discussed in the ROS Infrastructure PMC meeting today.
What came out of it is, the Infra team is totally ok with this change. The ROS team is up for it as long as we keep the name distinct enough to not make it seem like it comes from the ros_environment package.


```suggestion
ENV ROSDISTRO_PKGS_SYNC_DATE=@(ros_sync_date)@
```
I'm suggesting ROSDISTRO_PKGS_SYNC_DATE, if I get 2 upvotes and no strong no s I'll move forward with this change

@clalancette @nuclearsandwich @ruffsl @gavanderhoorn @tfoote FYI